### PR TITLE
Persistent notification

### PR DESCRIPTION
**Pull request by cammace**
Closes https://github.com/mapbox/mapbox-navigation-android/issues/174

Adds and updates a persistent notification during a navigation session. The current implementation can only be disabled and isn't customizable in any way (though I'll be changing this in a future release). This notification is separate from the drop-down notification on milestone trigger event which will be added in the future too.

Simple remoteViews layout:

Expanded remoteViews layout:

- [ ] Fix up image resources being different sizes
- [ ] document and clean up classes
- [ ] test

### CODE REVIEW

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationNotification.java [hunk 1]
```diff
@@ -20,43 +28,114 @@
...
+      }
+    }
+    remoteViews.setTextViewText(R.id.notificationStepDistanceTextView, distanceSpannableStringBuilder);
+
+    String arrivalTime = String.format(Locale.US, "Arrive at %s", formatArrivalTime(routeProgress.durationRemaining()));
+    remoteViews.setTextViewText(R.id.estimatedArrivalTimeTextView, arrivalTime);
+    remoteViewsBig.setTextViewText(R.id.estimatedArrivalTimeTextView, arrivalTime);
+
+    remoteViews.setImageViewResource(R.id.maneuverSignal,
+      ManeuverUtils.getManeuverResource(routeProgress.currentLegProgress().currentStep()));
```
Comment (danesfeder) [inline navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationNotification.java]: Should the maneuver image be retrieved with the upcoming step?

File: navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
-
 <LinearLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (danesfeder) [inline navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml]: Can these attributes be pushed to the `LinearLayout` below (and then remove this outermost parent)?  The child is one `LinearLayout`

File: navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml [hunk 3]
```diff
@@ -12,49 +11,60 @@
...
         <ImageView
-            android:id="@+id/imagenotileft"
-            android:layout_width="wrap_content"
+            android:id="@+id/maneuverSignal"
+            android:layout_width="64dp"
             android:layout_height="match_parent"
-            android:layout_margin="16dp"
             android:contentDescription="@string/app_name"
-            android:src="@drawable/ic_maneuver_starting"
+            android:src="@drawable/ic_maneuver_arrive"
```
Comment (danesfeder) [inline navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml]: I think it's better practice to load vector resources with `app:srcCompat` https://stackoverflow.com/questions/40624554/android-what-is-the-difference-between-appsrccompat-and-androidsrc

File: navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml (hunk 2 above)
Comment (danesfeder) [inline navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml]: Possible to move these attributes to the `LinearLayout` below and then remove?  The child of this parent is another `LinearLayout` and removing the parent would prevent another pass for drawing

File: navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml (hunk 3 above)
Comment (cammace) [reply navigation/libandroid-navigation/src/main/res/layout/layout_notification_default.xml]: Don't even need a resource here, accidentally added when designing the layout and I have now removed it.

**Pull request merged**